lib/WUI/link_content/prusa_link_api.cpp

```diff
@@ -89,18 +89,52 @@ optional<ConnectionState> PrusaLinkApi::accept(const RequestParser &parser) cons
              * transformation into a file name (removal of query params,
              * forbidding of '..', etc).
              */
+            /*
+             * TODO:
+             * We currently don't read the URI parameters. That is, we don't
+             * understand "?recursive=true" and therefore don't do recursive.
+             *
+             * The FileInfo handler doesn't know recursive yet either.
+             */
             static const auto prefix = "/api/files";
             static const size_t prefix_len = strlen(prefix);
             char fname[FILE_PATH_BUFFER_LEN + prefix_len];
             if (parser.uri_filename(fname, sizeof fname)) {
                 size_t len = strlen(fname);
-                const char *fname_real;
                 if (prefix_len > len) {
                     return StatusPage(Status::NotFound, parser.can_keep_alive());
-                } else if (prefix_len == len) {
-                    fname_real = "/";
-                } else {
-                    fname_real = fname + prefix_len;
+                }
+
+                const char *fname_real = fname + prefix_len;
+
+                /*
+                 * The octoprint API gives special meaning to /local and
+                 * /sdcard. For us, everything lives in the USB (/usb). We
+                 * remap these. Nevertheless, we never _generate_ these /local
+                 * or such URIs, so we don't remap anything else but the "root".
+                 */
+                static const char *const roots[] = {
+                    "",
+                    "/",
+                    "/local",
+                    "/local/",
+                    "/sdcard",
+                    "/sdcard/",
+                };
+
+                for (size_t i = 0; i < sizeof roots / sizeof *roots; i++) {
+                    if (strcmp(fname_real, roots[i]) == 0) {
```

Comment:
in this case `strcmp` is safe (`strncmp` not necessary) because we are comparing to statically defined strings in CPUFLASH

Reply:
`strncmp` would actually _not work_. I want exact match of the filename, not a prefix (in either direction). These are the paths that are „special“ in the octoprint API, but the sub-paths are never generated by us and I believe the client shall not make them up (let's see about that).

Reply:
I meant since one of the strings is constant, there is no danger of running past the buffer